src/client/components/vision/camera/view_model.ts

```diff
@@ -0,0 +1,282 @@
+import { observable } from 'mobx'
+import { computed } from 'mobx'
+import { createTransformer } from 'mobx-utils'
+import { BufferGeometry } from 'three'
+import { PlaneBufferGeometry } from 'three'
+import { RawShaderMaterial } from 'three'
+import { MeshBasicMaterial } from 'three'
+import { WebGLRenderTarget } from 'three'
+import { Scene } from 'three'
+import { Mesh } from 'three'
+import { WebGLRenderer } from 'three'
+import { Camera } from 'three'
+import { Object3D } from 'three'
+import { OrthographicCamera } from 'three'
+import { Material } from 'three'
+import { DataTexture } from 'three'
+import { LuminanceFormat } from 'three'
+import { RGBFormat } from 'three'
+import { Texture } from 'three'
+import { UnsignedByteType } from 'three'
+import { ClampToEdgeWrapping } from 'three'
+import { LinearFilter } from 'three'
+import { NearestFilter } from 'three'
+import { PlaneGeometry } from 'three'
+import { Matrix4 } from 'three'
+import { Vector4 } from 'three'
+import { Vector2 } from 'three'
+import { Geometry } from 'three'
+
+import { memoize } from '../../../base/memoize'
+
+import { CameraModel } from './model'
+import { ImageModel } from './model'
+import * as bayerFragmentShader from './shaders/bayer.frag'
+import * as bayerVertexShader from './shaders/bayer.vert'
+
+export function fourcc(code: string): number {
+  return code.charCodeAt(3) << 24 | code.charCodeAt(2) << 16 | code.charCodeAt(1) << 8 | code.charCodeAt(0)
+}
+
+export class CameraViewModel {
+  @observable.ref canvas: HTMLCanvasElement | null = null
+
+  constructor(private model: CameraModel) {
+  }
+
+  // See this createTransformer here? that means that once it is no longer observed
+  // it will be deleted. However the model that creates it will not.
+  // This means that any @computed fields will only live so long as the view does as
+  // this viewModel will only live so long as the view does.
```

Comment:
This comment seems like generic mobx documentation, which indicates this isn't the place to put it.

I think we can either remove or add it to a wiki/readme somewhere.

Reply:
I'm trying to explain that even though the @computed fields don't have a disposer they are ok since they are attached to the viewmodel/view. And when the viewmodel/view dies they will also be cleaned up.

Reply:
I'll change the comment to say what I meant to say :P

Reply:
On second thought it's really not needed at all